pkg/types/validation/installconfig_test.go

```diff
@@ -0,0 +1,299 @@
+			installConfig: validInstallConfig(),
+			valid:         true,
+		},
+		{
+			name: "missing name",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.ObjectMeta.Name = ""
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "missing cluster ID",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.ClusterID = ""
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "invalid ssh key",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.SSHKey = "bad-ssh-key"
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "invalid base domain",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.BaseDomain = ".bad-domain."
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "invalid network type",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Networking.Type = "bad-type"
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "invalid service cidr",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Networking.ServiceCIDR = ipnet.IPNet{}
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "invalid cluster network cidr",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Networking.ClusterNetworks[0].CIDR = "bad-cidr"
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "overlapping cluster network cidr",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Networking.ClusterNetworks[0].CIDR = "10.0.0.0/24"
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "cluster network host subnet length too large",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Networking.ClusterNetworks[0].CIDR = "192.168.1.0/24"
+				c.Networking.ClusterNetworks[0].HostSubnetLength = 9
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "missing master machine pool",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Machines = []types.MachinePool{
+					{
+						Name: "worker",
+					},
+				}
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "missing worker machine pool",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Machines = []types.MachinePool{
+					{
+						Name: "master",
+					},
+				}
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "duplicate machine pool",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Machines = append(c.Machines, types.MachinePool{
+					Name: "master",
+				})
+				return c
+			}(),
+			valid: false,
+		},
+		{
+			name: "invalid machine pool",
+			installConfig: func() *types.InstallConfig {
+				c := validInstallConfig()
+				c.Machines = append(c.Machines, types.MachinePool{
+					Name: "other",
```

Comment:
I think we want to support this.  See [here][1].

[1]: https://github.com/openshift/installer/issues/264#issuecomment-422349518

Reply:
hive team is going to be making that change soon, they can change the tests.
any right now we have nothing that supports adding these random pools.